File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    
    # 체육복 있는 학생
    s = set(lost) & set(reserve)
    
    # 도난 당해서 체육복 없는 학생
    plz_safe = set(lost) - s
    
    # 도난 당했으나 여벌 체육복을 가지고 있는 학생
    safe = set(reserve) - s
    
    for i in safe:
        if i - 1 in plz_safe:
            plz_safe.remove(i - 1)
        elif i + 1 in plz_safe:
            plz_safe.remove(i + 1)
        
    return n - len(plz_safe)

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n - lost.length;
        
        Arrays.sort(lost);
        Arrays.sort(reserve);
        
        for (int i = 0; i < reserve.length; i++) {
            for (int j = 0; j < lost.length; j++) {
                if (reserve[i] == lost[j]) {
                    answer++; // 이 학생은 여분이 있으므로 참여 가능
                    lost[j] = -1; // 불가 표시
                    reserve[i] = -1; // 불가 표시
                    break;
                }
            }
        }
        
        // 자기 번호로부터 +1, -1인 것만 가능
        for (int i = 0; i < lost.length; i++) {
            for (int j = 0; j < reserve.length; j++) {
                if (lost[i] - 1 == reserve[j] || lost[i] + 1 == reserve[j]) {
                    answer++;
                    reserve[j] = -1; // 빌려줌 표시
                    break;
                }
            }
        }
        
        return answer;
    }
}